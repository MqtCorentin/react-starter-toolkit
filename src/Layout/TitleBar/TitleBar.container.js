import React from 'react';
// eslint-disable-next-line import/no-unresolved
import { ClassManager } from '@axa-fr/react-toolkit-core';
import TitleBar from './TitleBar';

export default ({ backHome, classModifier, ...rest }) => {
  const newClassName = 'af-title-bar';
  const newClassModifier = `${classModifier || ''}${backHome ? ' backhome' : ''}`;
  const classComponent = ClassManager.getComponentClassName(newClassName, newClassModifier);
  return <TitleBar backHome={backHome} {...rest} className={classComponent} />;
};
