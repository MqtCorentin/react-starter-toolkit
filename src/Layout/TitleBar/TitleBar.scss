@import '@axa-fr/react-toolkit-layout-header/dist/Title/title-bar.scss';

.af-title-bar {
  margin-bottom: 0;
  &--backhome {
    .af-title-bar {
      &__wrapper {
        justify-content: flex-start;
        .af-title-bar__title {
          margin-left: 1em;
        }
      }
    }
  }
  &--oasis{
    margin-bottom: 2rem;
  }
}
