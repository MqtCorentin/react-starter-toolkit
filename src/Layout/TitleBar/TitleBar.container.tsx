import { ComponentProps } from 'react';
import TitleBar, { DEFAULT_CLASSNAME } from './TitleBar';

type TTitleBarContainer = ComponentProps<typeof TitleBar> & {
  TitleBarCmpt?: typeof TitleBar;
};

const TitleBarContainer = ({ backHome = false, classModifier = '', className = DEFAULT_CLASSNAME, ...rest }: TTitleBarContainer) => {
  const newClassModifier = [classModifier, backHome ? 'backhome' : ''].join(' ').trim();
  return <TitleBar {...rest} backHome={backHome} classModifier={newClassModifier} className={className} />;
};

export default TitleBarContainer;
