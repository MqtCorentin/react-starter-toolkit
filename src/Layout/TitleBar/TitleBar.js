import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import './TitleBar.scss';

const TitleBar = ({ backHome, children, className }) => (
  <div className={className}>
    <div className="container-fluid container af-title-bar__wrapper">
      {backHome && (
        <Link title="Retour Ã  l'accueil" className="btn af-btn--circle" to="/">
          <i className="glyphicon glyphicon-home" />
        </Link>
      )}
      <h1 className="af-title-bar__title">{children}</h1>
    </div>
  </div>
);

TitleBar.defaultProps = {
  backHome: false,
  children: null,
  className: 'af-title-bar',
};

TitleBar.propTypes = {
  backHome: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
};

export default TitleBar;
