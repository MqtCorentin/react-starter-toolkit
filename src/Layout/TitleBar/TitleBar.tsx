import { ReactNode } from 'react';
import { Link } from 'react-router-dom';
import { Title } from '@axa-fr/react-toolkit-all';
import { withClassDefault, withClassModifier, WithClassModifierOptions, compose, identity } from '@axa-fr/react-toolkit-core';
import './TitleBar.scss';

type TTitleBar = WithClassModifierOptions & {
  backHome?: boolean;
  children?: ReactNode;
  className?: string;
  title?: ReactNode;
  handleClick?: () => void;
  id?: string;
};

export const DEFAULT_CLASSNAME = 'af-title-bar';

const TitleBar = ({ title = 'title', handleClick, children = <></>, backHome = false, id = '', className }: TTitleBar) => (
  <Title id={id} className={className} toggleMenu={handleClick} title={title}>
    {backHome && (
      <Link aria-label="Retour Ã  l'accueil" className="btn af-btn--circle" to="/">
        <i className="glyphicon glyphicon-home" />
      </Link>
    )}
    {children}
  </Title>
);

const enhance = compose(identity<TTitleBar>(), withClassDefault(DEFAULT_CLASSNAME), withClassModifier());

export default enhance(TitleBar);
