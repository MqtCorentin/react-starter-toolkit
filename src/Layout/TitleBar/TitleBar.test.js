import React from 'react';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import TitleBar from './TitleBar';

const wrapper = ({ children }) => <StaticRouter context={{}}>{children}</StaticRouter>;

describe('<TitleBar/>', () => {
  it('Render <TitleBar />', () => {
    const { asFragment } = render(<TitleBar />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should not contain <Link /> when backHome is false', () => {
    const { getByTitle } = render(<TitleBar backHome />, { wrapper });
    expect(getByTitle("Retour Ã  l'accueil")).toBeDefined();
  });
});
