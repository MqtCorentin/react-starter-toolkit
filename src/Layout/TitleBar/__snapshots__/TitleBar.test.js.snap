// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar/> Render <TitleBar /> 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      />
    </div>
  </div>
</DocumentFragment>
`;
